### PR TITLE
`claimTo` for Drop and BundleDrop

### PR DESCRIPTION
**Pull request by ayshptk**

### CODE REVIEW

**Comment by jakeloo** [comment]
@ayshptk let's not change the interface. claim should keep default to 0s

File: src/modules/bundleDrop.ts [hunk 1]
```diff
@@ -463,7 +463,51 @@ export class BundleDropModule
...
+  ) {
+    const claimData = await this.prepareClaim(tokenId, quantity, proofs);
+
+    await this.sendTransaction(
+      "claim",
+      [tokenId, quantity, claimData.proofs],
+      claimData.overrides,
+    );
+  }
+  public async claimTo(
```
Comment (atbe) [inline src/modules/bundleDrop.ts:485]: docstring??

File: src/modules/bundleDrop.ts [hunk 2]
```diff
@@ -411,7 +411,7 @@ export class BundleDropModule
     await this.sendTransaction("setClaimConditions", [tokenId, _conditions]);
   }
 
-  public async claim(
+  private async prepareClaim(
```
Comment (atbe) [inline src/modules/bundleDrop.ts:414]: docstring??

File: src/modules/drop.ts [hunk 3]
```diff
@@ -659,10 +659,10 @@ export class DropModule
     );
   }
 
-  public async claim(
+  private async prepareClaim(
```
Comment (atbe) [inline src/modules/drop.ts:662]: docstring??

File: src/modules/drop.ts [hunk 4]
```diff
@@ -715,11 +715,40 @@ export class DropModule
         }
       }
     }
+    return {
+      overrides,
+      proofs,
+    };
+  }
 
+  public async claimTo(
```
Comment (atbe) [inline src/modules/drop.ts:724]: docstring??

**Comment by atbe** [comment]
Lets add tests for these new methods.

**Comment by ayshptk** [comment]
@atbe added

File: src/modules/bundleDrop.ts [hunk 5]
```diff
@@ -468,7 +474,77 @@ export class BundleDropModule
...
+   * @param data -
+   *
+   * @returns - Receipt for the transaction
+   */
+  public async claimTo(
+    tokenId: BigNumberish,
+    quantity: BigNumberish,
+    addressToClaim: string,
+    proofs: BytesLike[] = [hexZeroPad([0], 32)],
+    data: BytesLike = [0],
```
Comment (jakeloo) [inline src/modules/bundleDrop.ts:523]: we don't need to  expose this. hard it to `[0]` in the param.

File: src/modules/drop.ts [hunk 6]
```diff
@@ -664,10 +664,16 @@ export class DropModule
...
+   * Returns proofs and the overrides required for the transaction.
+   *
+   * @returns - `overrides` and `proofs` as an object.
+   */
+
+  private async prepareClaim(
     quantity: BigNumberish,
     proofs: BytesLike[] = [hexZeroPad([0], 32)],
-  ): Promise<NFTMetadataOwner[]> {
+  ): Promise<any> {
```
Comment (jakeloo) [inline src/modules/drop.ts:676]: why any here?

**Review by jakeloo — CHANGES_REQUESTED** [review]

File: src/modules/bundleDrop.ts [hunk 7]
```diff
@@ -468,7 +474,76 @@ export class BundleDropModule
...
+  }
+
+  /**
+   * Claim a token and send it to someone else
+   *
+   * @param tokenId - Id of the token you want to claim
+   * @param quantity - Quantity of the tokens you want to claim
+   * @param addressToClaim - Address you want to send the token to
+   * @param proofs - Array of proofs
+   * @param data -
```
Comment (jakeloo) [inline src/modules/bundleDrop.ts:514]: remove outdated docstring

**Review by jakeloo — APPROVED** [review]
you should be good, just minor clean up on docstring

**Pull request merged**